src/components/form/fields/schema-field.ts

```diff
@@ -0,0 +1,157 @@
+import { LimeElementsAdapter } from '../adapters';
+import JSONSchemaField from 'react-jsonschema-form/lib/components/fields/SchemaField';
+import React from 'react';
+import { FieldProps } from './types';
+import { isEmpty, capitalize } from 'lodash-es';
+
+const hasCustomComponent = (schema): boolean => {
+    return Boolean(schema.lime?.component?.name);
+};
+
+const getCustomComponent = (
+    schema
+): { name: string; props: { [key: string]: any } } => {
+    const name = schema.lime?.component?.name;
+    const props = schema.lime?.component?.props || {};
+
+    return { name: name, props: props };
+};
+
+export class SchemaField extends React.Component<FieldProps> {
```

Comment:
This component has a lot of the same functions as WidgetAdapter and looks very similar in many cases. However, most of the functions work slightly differently due to the differences between a widget and a field in rjsf. I thought about maybe trying to abstract away the similarities but it seemed excessive for a one off use case and I think it would have a negative output on the complexity and readability of the code. Just wanted to point that out since i figured it would be mentioned in code review :)